src/radical/saga/adaptors/shell/shell_job.py

```diff
@@ -568,6 +568,11 @@ def initialize(self):
         # stdio.
 
         base = self.base_workdir
+        with self._shell_lock:
+            ret, out, _ = self.shell.run_sync(' mkdir -p %s' % base)
+
+        if ret != 0:
+            raise rse.NoSuccess('host setup failed (%s):(%s)' % (ret, out))
```

Comment:
shouldn't the `mkdir` covered by `ru.get_radical_base()`?

Reply:
that is done for base dirs (inside `ru.get_radical_base()`), not for path added after,
kinda thought that having `get_radical_base('saga.adaptors.shell_job')` could be confusing, but I'm ok if you would want to switch to this format
and another thing, since here is used `self.shell.run_sync` (synced command execution), I wasn't sure should we switch from it (but also ok to switch considering to my comment above)

Reply:
Ah, of course - thanks.  Yeah, this LGTM then :-)